### PR TITLE
[YUNIKORN-1006] add notes about incompatible changes in the release doc

### PR DESCRIPTION
**Pull request by yangwwei**
Fix QU40

The project puts a high priority on backwards compatibility and aims to document any incompatible changes and provide tools and documentation to help users transition to new features.

### CODE REVIEW

File: docs/release-procedure.md [hunk 1]
```diff
@@ -74,6 +74,13 @@ Follow these steps to generate the list:
 - Select the style `Text` and click `create`
 - Scroll to the bottom of the page and copy the content of the text area and update the [CHANGELOG](../release-top-level-artifacts/CHANGELOG) file.
 
+## Incompatible Changes
+YuniKorn community puts a high priority to the backwards compatibility. For each release, all incompatible
```
Comment (yuchaoran2011) [inline docs/release-procedure.md:78]:
```suggestion
YuniKorn community puts a high priority on backward compatibility. For each release, all incompatible
```

File: docs/release-procedure.md [hunk 2]
```diff
@@ -74,6 +74,13 @@ Follow these steps to generate the list:
...
+changes must be carefully reviewed and documented. In general, incompatible changes to the public APIs usually requires
```
Comment (yuchaoran2011) [inline docs/release-procedure.md:79]:
```suggestion
changes must be carefully reviewed and documented. In general, incompatible changes to the public APIs usually require
```

File: docs/release-procedure.md [hunk 3]
```diff
@@ -74,6 +74,13 @@ Follow these steps to generate the list:
...
+a major version bump, this needs to be discussed in the private list. Examples are REST API changes, scheduler-interface
```
Comment (yuchaoran2011) [inline docs/release-procedure.md:80]:
```suggestion
a major version bump and they need to be discussed in the private list. Examples are REST API changes, scheduler-interface
```

File: docs/release-procedure.md [hunk 4]
```diff
@@ -74,6 +74,13 @@ Follow these steps to generate the list:
...
+main API changes. In the release notes, all incompatible changes must be clearly documented. User should be able to
```
Comment (yuchaoran2011) [inline docs/release-procedure.md:81]:
```suggestion
main API changes. In the release notes, all incompatible changes must be clearly documented. Users should be able to
```

**Review by yuchaoran2011 — CHANGES_REQUESTED** [review]
Minor nits. Otherwise looking good

**Comment by yangwwei** [comment]
thanks for the quick review @yuchaoran2011

**Review by yuchaoran2011 — APPROVED** [review]

**Pull request merged**